packages/flutter/lib/src/widgets/router.dart

```diff
@@ -0,0 +1,1159 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'binding.dart';
+import 'framework.dart';
+import 'navigator.dart';
+import 'route_notification_messages.dart';
+
+/// A piece of routing information
```

Comment:
nit: end with `.`

This doc also needs to describe what the routing information is.